app/src/main/java/com/google/android/ground/ui/compose/ConfirmationDialog.kt

```diff
@@ -20,27 +20,58 @@ import androidx.compose.material3.AlertDialog
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextButton
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.google.android.ground.ExcludeFromJacocoGeneratedReport
 import com.google.android.ground.R
+import com.google.android.ground.ui.theme.AppTheme
 
 @Composable
 fun ConfirmationDialog(
   @StringRes title: Int,
   @StringRes description: Int,
-  @StringRes dismissButtonText: Int = R.string.cancel,
-  @StringRes confirmButtonText: Int,
-  signOutCallback: () -> Unit,
-  dismissCallback: () -> Unit,
+  @StringRes dismissButton: Int = R.string.cancel,
+  @StringRes confirmButton: Int,
+  onConfirmClicked: () -> Unit,
 ) {
-  AlertDialog(
-    onDismissRequest = { dismissCallback() },
-    title = { Text(text = stringResource(title)) },
-    text = { Text(text = stringResource(description)) },
-    dismissButton = {
-      TextButton(onClick = { dismissCallback() }) { Text(text = stringResource(dismissButtonText)) }
-    },
-    confirmButton = {
-      TextButton(onClick = { signOutCallback() }) { Text(text = stringResource(confirmButtonText)) }
-    },
-  )
+  val showRemoveWarningDialog = remember { mutableStateOf(true) }
+
+  fun onConfirm() {
+    showRemoveWarningDialog.value = false
+    onConfirmClicked()
+  }
+
+  fun onDismiss() {
+    showRemoveWarningDialog.value = false
+  }
+
+  if (showRemoveWarningDialog.value) {
+    AlertDialog(
+      onDismissRequest = { onDismiss() },
+      title = { Text(text = stringResource(title)) },
+      text = { Text(text = stringResource(description)) },
+      dismissButton = {
+        TextButton(onClick = { onDismiss() }) { Text(text = stringResource(dismissButton)) }
+      },
+      confirmButton = {
+        TextButton(onClick = { onConfirm() }) { Text(text = stringResource(confirmButton)) }
+      },
+    )
+  }
+}
+
+@Composable
+@Preview
+@ExcludeFromJacocoGeneratedReport
+fun PreviewConfirmationDialog() {
```

Comment:
just a suggestion, how about creating our own custom annotation which internally includes all these three annotations so that in future we have to add one annotation only and can prevent anyone forgetting exclude thing?

WDYT?

```kt
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
@Composable
@Preview
@ExcludeFromJacocoGeneratedReport
annotation class ComposablePreview
```


```kt
@ComposablePreview
fun PreviewConfirmationDialog() {
   ...
}
```

Reply:
Thanks for the suggestion! I would to keep `Preview` out of the proposed annotation as it gives us the flexibility to add other preview options which may or may not be applicable to all use cases (e.g. I don't like to add background to dialog, for single items i don't prefer showSystemUi so that multiple previews can be fit into one). Can we move this discussion to a new bug?

Reply:
Sure, SG